### PR TITLE
ROC-4406 Add priority debt to statement of means

### PR DESCRIPTION
**Pull request by ellie-harrison**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-4406

### Change description
Add priority debt page to the statement of means journey

### CODE REVIEW

File: src/main/features/response/views/statement-of-means/priority-debts.njk [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        {{ csrfProtection(csrf) }}
+
+        {{ priorityDebt(form, name = 'mortgage', label = PriorityDebtType.MORTGAGE.displayValue | capitalize) }}
+        {{ priorityDebt(form, name = 'rent', label = PriorityDebtType.RENT.displayValue | capitalize) }}
+        {{ priorityDebt(form, name = 'councilTax', label = PriorityDebtType.COUNCIL_TAX_COMMUNITY_CHARGE.displayValue | capitalize) }}
+        {{ priorityDebt(form, name = 'gas', label = PriorityDebtType.GAS.displayValue | capitalize) }}
+        {{ priorityDebt(form, name = 'electricity', label = PriorityDebtType.ELECTRICITY.displayValue | capitalize) }}
+        {{ priorityDebt(form, name = 'water', label = PriorityDebtType.WATER.displayValue | capitalize) }}
+        {{ priorityDebt(form, name = 'maintenance', label = PriorityDebtType.MAINTENANCE_PAYMENTS.displayValue | capitalize) }}
+
```
Comment (ckalista) [inline src/main/features/response/views/statement-of-means/priority-debts.njk]: Extra un-needed space.

File: src/main/features/response/views/statement-of-means/priority-debts.njk [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+{% macro priorityDebt(form, name, label) %}
+  {% set checkboxName = name + 'Declared' %}
+
+  {% call expandableCheckboxOption(form = form, name = checkboxName, value = 'true', checked = (form.valueFor(checkboxName) or (form.valueFor(name) and form.valueFor(name).populated)), label = label) %}
+    {{ incomeExpenseSource(form = form, name = name, sourceLabel = 'Source of income', amountLabel = 'Repayment Amount', amountHint = 'Only include the amount you’re paying to clear previously missed regular payments', scheduleLabel = 'Repaid every') }}
+    {{ linkButton(name = 'action[resetDebt][' + name + ']', label = 'Reset this debt', class = 'link-button js-hidden') }}
+  {% endcall %}
+{% endmacro %}
+
```
Comment (ckalista) [inline src/main/features/response/views/statement-of-means/priority-debts.njk]: Extra un-needed space.

File: src/test/features/response/form/models/statement-of-means/priorityDebt.ts [hunk 3]
```diff
@@ -0,0 +1,224 @@
+import { Validator } from 'class-validator'
+import { PriorityDebt } from 'response/form/models/statement-of-means/priorityDebt'
+import { PriorityDebtType } from 'response/form/models/statement-of-means/priorityDebtType'
+import { IncomeExpenseSchedule } from 'response/form/models/statement-of-means/incomeExpenseSchedule'
+import { expect } from 'chai'
+import { expectValidationError } from '../../../../../app/forms/models/validationUtils'
```
Comment (sabahirfan) [inline src/test/features/response/form/models/statement-of-means/priorityDebt.ts]: please remove `../../../../../`

File: src/main/app/claims/responseModelConverter.ts [hunk 4]
```diff
@@ -338,6 +341,66 @@ export class ResponseModelConverter {
     return children
   }
 
+  private static convertPriorityDebts (priorityDebt: PriorityDebtDraft | undefined): PriorityDebts[] {
```
Comment (sabahirfan) [inline src/main/app/claims/responseModelConverter.ts]: Don't see any test to cover this

**Pull request closed without merging**